### PR TITLE
Set NumProc on tizen

### PR DESCRIPTION
**Pull request by gbalykov**
cc @alpencolt

### CODE REVIEW

**Comment by jkotas** [comment]
Was the fix made in https://github.com/dotnet/runtime/pull/51002 incomplete?

File: eng/native/build-commons.sh [hunk 1]
```diff
@@ -289,6 +289,17 @@ else
   fi
 fi
 
+if [[ "$platform" == "Linux" && "$__NumProc" == "1" ]]; then
```
Comment (am11) [inline eng/native/build-commons.sh:292]: nit: for consistency, we can move it after line 286 (above `else\n__NumProc=1` block) and use same single-line condition which we used here https://github.com/dotnet/runtime/blob/0dd6b7be33b9a3bd5953bceff07377839af92a5c/eng/native/init-os-and-arch.sh#L45

**Comment by gbalykov** [comment]
@jkotas #51002 was a more common build fix in case there's no nproc, maybe this can happen not only on tizen. Otherwise, build will fail. This PR enables concurrent build specifically on tizen.

File: eng/native/build-commons.sh (hunk 1 above)
Comment (gbalykov) [reply eng/native/build-commons.sh]: Thanks!

**Review by jkotas — APPROVED** [review]

**Pull request merged**